plugin/src/withAuth0.ts

```diff
@@ -0,0 +1,156 @@
+  }
+  if (!auth0Scheme) {
+    throw Error('No auth0 scheme specified in expo config');
+  }
+  return mergeContents({
+    tag: 'react-native-auth0-manifest-placeholder',
+    src,
+    newSrc: `manifestPlaceholders = [auth0Domain: "${auth0Domain}", auth0Scheme: "${auth0Scheme}"]`,
+    anchor: /defaultConfig {/,
+    offset: 1,
+    comment: '//',
+  }).contents;
+};
+
+const withAndroidAuth0Gradle: ConfigPlugin<Auth0PluginConfig | void> = (
+  config,
+  {android, scheme, domain} = {},
+) => {
+  return withAppBuildGradle(config, config => {
+    if (config.modResults.language === 'groovy') {
+      const auth0Domain =
+        process.env.EXPO_AUTH0_DOMAIN_ANDROID ||
+        process.env.EXPO_AUTH0_DOMAIN ||
+        android?.domain ||
+        domain;
+      const auth0Scheme =
+        process.env.EXPO_AUTH0_SCHEME_ANDROID ||
+        process.env.EXPO_AUTH0_SCHEME ||
+        android?.scheme ||
+        scheme ||
+        '${applicationId}';
+
+      config.modResults.contents = addAuth0GradleValues(
+        config.modResults.contents,
+        auth0Domain,
+        auth0Scheme,
+      );
+      return config;
+    } else {
+      throw new Error(
+        'Cannot add auth0 build.gradle modifications because the build.gradle is not groovy',
+      );
+    }
+  });
+};
+
+export const addAuth0AppDelegateCode = (src: string): string => {
+  let tempSrc = src;
+  // Tests to see if the RCTLinkingManager has already been added
+  if (
+    !/\[RCTLinkingManager.*application:.*openURL:.*options:.*\]/.test(tempSrc)
+  ) {
+    tempSrc = mergeContents({
+      src: tempSrc,
+      newSrc: [
+        '- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url',
+        '            options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options',
+        '{',
+        '  return [RCTLinkingManager application:app openURL:url options:options];',
+        '}',
+      ].join('\n'),
+      tag: 'react-native-auth0-linking',
+      anchor: /@end/,
+      comment: '//',
+      offset: 0,
+    }).contents;
+  }
+  // Checks to see if RCTLinkingManager hasn't been imported
+  if (!/RCTLinkingManager\.h/.test(tempSrc)) {
+    tempSrc = mergeContents({
+      src: tempSrc,
+      newSrc: `#import <React/RCTLinkingManager.h>`,
+      anchor: /#import <React\/RCTBridge\.h>/,
+      offset: 1,
+      tag: 'react-native-auth0-import',
+      comment: '//',
+    }).contents;
+  }
+  return tempSrc;
+};
+
+const withIOSAuth0AppDelegate: ConfigPlugin<Auth0PluginConfig | void> = config => {
+  return withAppDelegate(config, config => {
+    const src = config.modResults.contents;
+    config.modResults.contents = addAuth0AppDelegateCode(src);
+    return config;
+  });
+};
+
+const withIOSAuth0InfoPList: ConfigPlugin<Auth0PluginConfig | void> = (
+  config,
+  {ios, scheme} = {},
+) => {
+  return withInfoPlist(config, config => {
+    if (!process.env.EXPO_AUTH0_DISABLE_PLIST_MOD && !ios?.disablePListMod) {
```

Comment:
Probably you can do away with this in favor of only enabling the feature when a scheme is defined. If users _really_ want to go against the required functionality, they can add another config plugin removing schemes.